authors/apps/authentication/backends.py

```diff
@@ -1,14 +1,55 @@
-# import jwt
-#
-# from django.conf import settings
-#
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
+import jwt
 
-"""Configure JWT Here"""
+from django.conf import settings
 
+from rest_framework import authentication, exceptions
 
-class JWTAuthentication:
-    def authenticate(self, another):
-        pass
+from .models import User
+
+
+class JWTAuthentication(authentication.BaseAuthentication): # NOQA
+    """ JWTAuthenticattion implement authentication
+        of token  received from the 'Authorization' Headers by
+        the keyword 'Token'"""
+
+    # initializing the token as an empty string
+    keyword = 'Token'
+
+    def authenticate(self, request):
+        request.user = None
+
+        # returns Authorization header
+        # as a bytestring
+
+        auth_header = authentication.get_authorization_header(request).split()
+
+        if not auth_header or auth_header[0].decode().lower() != self.keyword.lower():
+            return None
+
+        if len(auth_header) == 1:
+                raise exceptions.AuthenticationFailed('Invalid token header. No credentials provided.')
+        elif len(auth_header) > 2:
+            raise exceptions.AuthenticationFailed('Invalid token header. Token string should not contain spaces.')
+        return self.authenticate_credentials(request, auth_header[1].decode())
+
+    def authenticate_credentials(self, request, token): # NOQA
+        """
+        authenticate the given credentials. If authentication is
+        successful, return the user and token. If not, throw an error.
+        """
+        try:
+            payload = jwt.decode(token, settings.SECRET_KEY)
+        except (jwt.DecodeError, jwt.ExpiredSignatureError) as e:
+            raise exceptions.AuthenticationFailed(str(e))
```

Comment:
This isn't what I advised you to do @YAHYA-H . The raised exception message won't be descriptive.